pkg/skaffold/deploy/helm/util.go

```diff
@@ -0,0 +1,243 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helm
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/constants"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+	"github.com/blang/semver"
+	"github.com/sirupsen/logrus"
+)
+
+func IsHelmChart(path string) bool {
+	return filepath.Base(path) == "Chart.yaml"
+}
+
+// copy of cmd/skaffold/app/flags.BuildOutputs
+type buildOutputs struct {
+	Builds []build.Artifact `json:"builds"`
+}
+
+func writeBuildArtifacts(builds []build.Artifact) (string, func(), error) {
+	buildOutput, err := json.Marshal(buildOutputs{builds})
+	if err != nil {
+		return "", nil, fmt.Errorf("cannot marshal build artifacts: %w", err)
+	}
+
+	f, err := ioutil.TempFile("", "builds*.yaml")
+	if err != nil {
+		return "", nil, fmt.Errorf("cannot create temp file: %w", err)
+	}
+	if _, err := f.Write(buildOutput); err != nil {
+		return "", nil, fmt.Errorf("cannot write to temp file: %w", err)
+	}
+	if err := f.Close(); err != nil {
+		return "", nil, fmt.Errorf("cannot close temp file: %w", err)
+	}
+	return f.Name(), func() { os.Remove(f.Name()) }, nil
+}
+
+// sortKeys returns the map keys in sorted order
+func sortKeys(m map[string]string) []string {
+	s := make([]string, 0, len(m))
+	for k := range m {
+		s = append(s, k)
+	}
+	sort.Strings(s)
+	return s
+}
+
+// binVer returns the version of the helm binary found in PATH. May be cached.
```

Comment:
If i had to nit pick, my only nit would be do place class methods in the same file as class definition or group the method definition and where its called in one file.  (To avoid jumping back and forth different files when reading code)  

Related [thread](https://softwareengineering.stackexchange.com/questions/199311/most-human-friendly-way-to-order-class-method-definitions)

However, we have not followed that pattern elsewhere especially with the runner class

Reply:
yeah that's a great point. in general our code organization in this package is bad at best and non-existent at worst, i've been poking around in here lately and it can be impossible to follow. i originally just intended to split out the methods that were being used in multiple places, but got a little trigger happy i think. in general though i'd be very happy if we could stick to this pattern everywhere in skaffold :) 

i'll be touching more of this code soon so i'll probably send a few more refactors like this, i'll keep this pattern in mind.